packages/public-api/src/clientApiRouter.ts

```diff
@@ -50,15 +55,24 @@ type RequestWebPageScanResponse = Pick<WebPageScan, 'status' | 'finishedAt'> & {
 
 export const appRouter = trpc
   .router<Context>()
-  .mutation('requestWebPageScan', {
+  .mutation('requestWebPageRescan', {
+    input: z.string().url(),
+    async resolve({ input: url }) {
+      return await requestWebPageRescan(url);
+    },
+  })
+  .query('getWebPageScan', {
     input: z.string().url(),
     async resolve({ input: url }) {
-      const scan = await getOrRequestWebPageScan(url);
+      const scan = await getWebPageScan(url);
+      if (!scan) {
+        return null;
```

Comment:
What's the reasoning behind returning a `null` instead of throwing a `NotFound`?